piksi_tools/console/settings_view.py

```diff
@@ -505,41 +475,82 @@ def display_ins_output_hint(self):
+            "    The software reset will temporarily interrupt the console's connection to the device but it   \n" \
+            "    will recover on its own.    \n\n"
 
         confirm_prompt2.run(block=False)
 
-    def _send_pending_settings_by_index(self):
-        for eachindex in self.pending_settings:
-            self.link(MsgSettingsReadByIndexReq(index=eachindex))
-
-    def _restart_retry_thread(self):
-        if self.retry_pending_read_index_thread:
-            self.retry_pending_read_index_thread.stop()
-        self.retry_pending_read_index_thread = TimedDelayStoppableThread(
-            SETTINGS_RETRY_TIMEOUT,
-            target=self._send_pending_settings_by_index, args=[])
-        self.retry_pending_read_index_thread.start()
-
-    def _settings_read_by_index(self):
-        self.enumindex = 0          # next index to ask for
-        self.pending_settings = []  # list of settings idices we've asked for
-        self.ordering_counter = 0   # helps make deterministic order of settings
-        self.setup_pending = True   # guards against receipt of multiple "done" msgs
-        # queue up BATCH_WINDOW settings indices to read
-        self.pending_settings = range(self.enumindex, self.enumindex + BATCH_WINDOW)
-        self.enumindex += BATCH_WINDOW
-        self._send_pending_settings_by_index()
-        # start a thread that will resend any read indexes that haven't come
-        self._restart_retry_thread()
+    def _settings_unconfirm_all(self):
+        for section in self.settings.keys():
+            for name in self.settings[section].keys():
+                self.settings[section][name].confirmed_set = False
+
+    def _settings_read_all(self):
+        self._settings_unconfirm_all()
+        threading.Thread(target=self._settings_read_all_thd).start()
+
+    def _settings_read_all_thd(self):
+        slist = self.s_api.read_all()
+
+        idx = 0
+
+        for s in slist:
+            section = s['section']
+            name = s['name']
+            value = s['value']
+            fmt_type = s['fmt_type']
+
+            idx += 1
+
+            if fmt_type == '':
+                setting_type = None
+                setting_format = None
+            else:
+                setting_type, setting_format = fmt_type.split(':')
+
+            if section not in self.settings:
+                self.settings[section] = {}
+
+            # setting exists, we won't reinitilize it but rather update existing setting
+            dict_setting = self.settings[section].get(name, False)
+            if dict_setting:
+                dict_setting.value = value
+                dict_setting.ordering = idx
+                if setting_type == 'enum':
+                    enum_values = setting_format.split(',')
+                    dict_setting.enum_values = enum_values
+                dict_setting.confirmed_set = True
+                continue
+
+            if setting_type is None:
```

Comment:
The `if` and `else` clause look the same, merge into `if setting_type != 'enum'`?

Reply:
Agreed.